src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+        assertEquals(leadingByte, bytes[0]);
+        ByteArraySet deSet = ByteArraySet.deserialize(bytes);
+
+        for(ByteArray val: vals) {
+            assertTrue(deSet.contains(val));
+        }
+        assertEquals(size, deSet.size());
+
+        int count = 0;
+        for(ByteArray val: deSet) {
+            assertTrue(vals.contains(val));
+            count++;
+        }
+        assertEquals(size, count);
+    }
+
+    @Test
+    public void emptySerializeDeserialize() {
+        testSerializeDeserialize(0, (byte) 0);
+    }
+
+    @Test
+    public void reallySmallSerializeDeserialize() {
+        testSerializeDeserialize(1, (byte) 1);
+    }
+
+    @Test
+    public void smallSerializeDeserialize() {
+        testSerializeDeserialize(2, (byte) 2);
+    }
+
+    @Test
+    public void regularSerializeDeserialize() {
+        testSerializeDeserialize(SIZE_REGULAR, (byte) 2);
+    }
+
+    @Test
+    public void bigSerializeDeserialize() {
+        testSerializeDeserialize(SIZE_BIG, (byte) 3);
+    }
+
+    @Test
+    public void reallyBigSerializeDeserialize() {
+        testSerializeDeserialize(SIZE_REALLY_BIG, (byte) 3);
+    }
+
+    private void testIterator(int size) {
+        System.out.println("Testing iterator method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+        assertEquals(size, set.size());
+
+        Iterator<ByteArray> iter = set.iterator();
+        List<ByteArray> alreadySeenVals = new ArrayList<>();
+        int count = 0;
+        while(iter.hasNext()) {
+            ByteArray val = iter.next();
+            assertFalse(alreadySeenVals.contains(val));
+            alreadySeenVals.add(val);
+            assertTrue(vals.contains(val));
+            count++;
+        }
+        assertEquals(size, count);
+    }
+
+    @Test
+    public void emptyIterator() {
+        testIterator(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallIterator() {
+        testIterator(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallIterator() {
+        testIterator(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularIterator() {
+        testIterator(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigIterator() {
+        testIterator(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigIterator() {
+        testIterator(SIZE_REALLY_BIG);
+    }
+
+    private void testRemove(int size) {
+        System.out.println("Testing remove method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
```

Comment:
```suggestion
        assertTrue(set.addAll(getRandomByteArrays(size)));
```
This should be identical. `ByteArraySet.addAll()` calls `ByteArraySet.add()` internally and will return False if it can't add all

Reply:
This can also probably be done multiple places in the tests outside of the explicit `ByteArraySet.add()` tests